### PR TITLE
[TypeChecker] Adjust witness matching to strip concurrency from ObjC …

### PR DESCRIPTION
**Pull request by xedin**
…requirements

`swift_attr` attributes in the type context were ignored before, 
which means that we need to maintain status quo to avoid breaking
witness matching by stripping everything concurrency related from inner types.

Resolves: rdar://122592751

### CODE REVIEW

**Comment by xedin** [comment]
@swift-ci please test

**Review by ktoso — APPROVED** [review]

**Review by DougGregor — APPROVED** [review]

File: lib/Sema/TypeCheckProtocol.cpp [hunk 1]
```diff
@@ -1169,6 +1169,33 @@ swift::matchWitness(WitnessChecker::RequirementEnvironmentCache &reqEnvCache,
   // Match a type in the requirement to a type in the witness.
   auto matchTypes = [&](Type reqType,
                         Type witnessType) -> llvm::Optional<RequirementMatch> {
+    // `swift_attr` attributes in the type context were ignored before,
+    // which means that we need to maintain status quo to avoid breaking
+    // witness matching by stripping everything concurrency related from
+    // inner types.
+    if (req->isObjC()) {
```
Comment (DougGregor) [inline lib/Sema/TypeCheckProtocol.cpp:1176]: Can we limit this stripping behavior to Swift 5 without strict concurrency?

Comment (xedin) [reply lib/Sema/TypeCheckProtocol.cpp]: Done!

**Comment by xedin** [comment]
@swift-ci please test

**Comment by xedin** [comment]
@swift-ci please test macOS platform

**Pull request merged**